stl/inc/compare

```diff
@@ -474,16 +476,15 @@ namespace _Weak_order {
                 return {_St::_None};
             } else if constexpr (_Has_ADL<_Ty1, _Ty2>) {
                 return {_St::_Adl, noexcept(static_cast<weak_ordering>(
-                                       /* ADL */ weak_order(_STD declval<_Ty1&>(), _STD declval<_Ty2&>())))};
+                                       weak_order(_STD declval<_Ty1&>(), _STD declval<_Ty2&>())))}; // intentional ADL
             } else if constexpr (floating_point<decay_t<_Ty1>>) {
                 return {_St::_Floating, true};
             } else if constexpr (_Can_compare_three_way<_Ty1, _Ty2>) {
                 return {_St::_Three, noexcept(static_cast<weak_ordering>(
                                          compare_three_way{}(_STD declval<_Ty1&>(), _STD declval<_Ty2&>())))};
             } else if constexpr (_Strong_order::_Has_ADL<_Ty1, _Ty2>) {
-                return {_St::_Strong, noexcept(static_cast<weak_ordering>(static_cast<strong_ordering>(
-                                          /* ADL, throughput optimization */ strong_order(
-                                              _STD declval<_Ty1&>(), _STD declval<_Ty2&>()))))};
+                return {_St::_Strong, noexcept(static_cast<weak_ordering>(static_cast<strong_ordering>(strong_order(
+                                          _STD declval<_Ty1&>(), _STD declval<_Ty2&>()))))}; // intentional ADL

```

Comment:
Here and below, a "throughput optimization" comment is being removed. This was intended to explain that we're taking an (invisible) shortcut compared to the Standard's depicted wording. Are you intentionally removing these comments? (I would prefer to keep this part, but of course I wrote them :joy_cat:)

Reply:
It was intentional, to get a bit of extra space so I can consistently use the `intentional ADL` comment. I think these are somewhat low-value given the more extensive "throughput optimization" comments on the unqualified name lookup blocks for these expressions. This is, admittedly, a harder argument to support for the case of lines 558, 627, or 630 where that other comment is quite far away.

I'll bring these back, I think as `// throughput optimization (see above):` on the lines preceding the `return` statements.